### PR TITLE
Fix #10411 default environment

### LINKED ISSUE #10411: Review deployment of Jetty Context XML files
**Jetty version(s)**
12

**Description**
Follow-up issue to #10158, where the deployment of Jetty Context XML files should be able to pick a default environment.
See the comments in #10158, as this should not be resolved by documentation only -- it should work out of the box.

### PR DESCRIPTION
**Pull request by gregw**
Fix #10411 
Implemented a simpler default environment algorithm where an application that does not specify an environment is always attempted in the default.

### CODE REVIEW

**Review by joakime — APPROVED** [review]
Seems simpler. +1

**Comment by gregw** [comment]
As this is ready, I have moved it back for consideration in 12.0.1

File: jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Deployable.java [hunk 1]
```diff
@@ -25,22 +24,31 @@
...
+            {
+                int n1 = Integer.parseInt(m1.group(1));
+                int n2 = Integer.parseInt(m2.group(1));
+                return Integer.compare(n1, n2);
+            }
+            return 1;
         }
-        return 0;
+        if (m2.matches())
+            return -11;
```
Comment (lorban) [inline jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Deployable.java]: Does this one _really_ has to go to 11?

**Review by lorban — CHANGES_REQUESTED** [review]
Either that's a typo, or we need a comment explaining the -11.

File: jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Deployable.java (hunk 1 above)
Comment (sbordet) [inline jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Deployable.java]: Should be -1?

**Review by sbordet — CHANGES_REQUESTED** [review]
Please go do file `deploy-jetty-adoc` line 40 and modify the sentence from:

`The Jetty content XML file must be accompanied...`

to 

`The Jetty content XML file may be accompanied...`

Also, modify `DistributionTests.testXmlDeployWarNotInWebapps()` so that it has 1 more boolean parameter, namely whether to write the `demo.properties` file or not.
In this way, we are testing that your fixes work as intended even in the case the `*.war` is not inside `$JETTY_HOME/webapps/`.

If you wan to go the extra mile, I would not mind in `DistributionTests` to add a test with 2 XML files and different environments that also verifies that one of them must have the `*.properties` file to be deployed correctly.

**Review by lorban — APPROVED** [review]

**Pull request merged**